### PR TITLE
Add Flipper support.

### PR DESCRIPTION
**Pull request by eddwk**
## Why?
https://github.com/smashingboxes/boxcar/issues/122
## What Changed?
-Add flipper to gemfile. 
-Add active record adapter for flipper. 
-Add templates for flipper and flipper-ui. 
-Add routing constraints for the UI in non-development environments.

### CODE REVIEW

File: templates/flipper.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Flipper.configure do |config|
+  config.default do
+   require 'flipper/adapters/active_record'
```
Comment (dkniffin) [inline templates/flipper.rb]: Is this `require` supposed to go inside the block here, or up top? It seems weird to see it down here.

File: templates/routes.rb.erb [hunk 2]
```diff
@@ -1,11 +1,32 @@
...
   mount_devise_token_auth_for 'User',
                               at: 'api/v1/users',
                               controllers: {
                                 sessions: "api/v1/users/sessions",
                               }
   <%- end -%>
+  
+  <%- if config[:flipper] -%>
+  if Rails.env.development? 
+  mount Flipper::UI.app(Flipper), at: '/flipper'
```
Comment (dkniffin) [inline templates/routes.rb.erb]: Should this be indented?

File: templates/routes.rb.erb [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
+  else
+  <%- if config[:devise] -%>
+  constraints FlipperRoutingConstraints do
```
Comment (dkniffin) [inline templates/routes.rb.erb:22]: Should this be indented?

File: templates/user.rb.erb [hunk 4]
```diff
@@ -3,5 +3,10 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+  <%- if config[:devise_token_auth] -%>
   include DeviseTokenAuth::Concerns::User
+  <%- end -%>
+  def admin?
```
Comment (dkniffin) [inline templates/user.rb.erb:11]: Can you take a look at the generated file here? I think there's going to be no space between `include DeviseTokenAuth::Concerns::User` and `def admin?`

File: lib/boxcar/app_builder.rb [hunk 5]
```diff
@@ -4,6 +4,7 @@
 # app. For example, the readme method is responsible for creating the readme file. Since we're
 # inheriting from Rails::AppBuilder, we have a lot of those building blocks set up for us already.
 # We can override those methods to customize how they work, or we can create new ones.
+
```
Comment (dkniffin) [inline lib/boxcar/app_builder.rb]: I think this got added accidentally. Let's remove it.

```suggestion
```

**Review by dkniffin — COMMENTED** [review]
Just a few small things

File: lib/boxcar/app_builder.rb (hunk 5 above)
Comment (eddwk) [reply lib/boxcar/app_builder.rb]: 👍

File: templates/user.rb.erb (hunk 4 above)
Comment (eddwk) [reply templates/user.rb.erb:11]: You are right, fixed.

File: templates/routes.rb.erb (hunk 3 above)
Comment (eddwk) [reply templates/routes.rb.erb:22]: 👍

File: templates/routes.rb.erb (hunk 2 above)
Comment (eddwk) [reply templates/routes.rb.erb]: 👍

File: templates/flipper.rb (hunk 1 above)
Comment (eddwk) [reply templates/flipper.rb]: 👍

File: lib/boxcar/app_builder.rb [hunk 6]
```diff
@@ -3,7 +3,7 @@
 # The purpose of this class is to provide the building blocks that are used for generating a rails
 # app. For example, the readme method is responsible for creating the readme file. Since we're
 # inheriting from Rails::AppBuilder, we have a lot of those building blocks set up for us already.
-# We can override those methods to customize how they work, or we can create new ones.
+# We can override those methods to customize how they work, or we can create new ones.=
```
Comment (dkniffin) [inline lib/boxcar/app_builder.rb]: You added an `=` here.
```suggestion
# We can override those methods to customize how they work, or we can create new ones.
```

File: templates/routes.rb.erb (hunk 3 above)
Comment (dkniffin) [reply templates/routes.rb.erb:22]: I think you forgot to address indenting this one

File: templates/routes.rb.erb [hunk 7]
```diff
@@ -1,11 +1,32 @@
...
+  <%- if config[:flipper] -%>
+  if Rails.env.development? 
+    mount Flipper::UI.app(Flipper), at: '/flipper'
+  else
+  <%- if config[:devise] -%>
+  constraints FlipperRoutingConstraints do
+    mount Flipper::UI.app(Flipper) => '/flipper'
+  end
+  <%- else -%>
+  # If you uncomment this line, Flipper UI will be completely unprotected.
```
Comment (dkniffin) [inline templates/routes.rb.erb:26]: I think this is supposed to be indented too.

File: lib/boxcar/app_builder.rb (hunk 6 above)
Comment (eddwk) [reply lib/boxcar/app_builder.rb]: !=

**Review by dkniffin — APPROVED** [review]
:shipit:

**Pull request merged**